### ISSUE TITLE
Unable to build while using await

### ISSUE THREAD

**Issue by mitow7821** [chromaui/chromatic-cli#376]
I'am trying to pass data that is result of await fetch to storybook arg (i use that data to customize final args), when I do so and try to run chromatic command this error occurs:
![image](https://user-images.githubusercontent.com/34492112/123953872-049c0900-d9a8-11eb-9541-22731100a27d.png)

When I run build-storybook and host it, everything works fine. 
How to make it work, while I can't use loaders to handle async operations?

**Comment by tmeasday** [comment]
Hi @mitow7821, does the storybook work when you do that? I would expect it would. We have loaders for this purpose (https://storybook.js.org/docs/vue/writing-stories/loaders) although loading story data async isn't really recommended.

**Comment by mitow7821** [comment]
How could I apply loader in this example where I get data with fetch and I NEED to pass it to get my args for stories that are defined dynamically. 
![image](https://user-images.githubusercontent.com/34492112/124074732-3662ad80-da44-11eb-82ed-1cf0568e8473.png)

**Comment by mitow7821** [comment]
I've tried to use loaders instead, final results are almost the same (i can access that data on storybook, test are failing because of asynchronous getting data, but while I am using my method chromatic doesnt want to build, which doesnt happend using loaders. 

Vue can handle async operation, with <Suspense> added to template, but chromatic is just blocking it from building without loader.

**Comment by mitow7821** [comment]
@tmeasday yes it is working on dev/prod, i am using vue `<suspense>` and it enables vue components to be loaded asynchronously.
I've tried to use loaders instead, final results are almost the same. I can access that data on storybook, but with chromatic my implementation doesnt want to build. While loader is used, verification step is passed (but i can't use loader in my code becasue i wouldn't have access to it's value in Primary.args = {} lines)

**Comment by tmeasday** [comment]
@mitow7821 perhaps you can write into support so we can look more closely at your uploaded builds.

**Comment by zernie** [comment]
I'm having the same issue with React + Suspense app previews. It works perfectly on dev but breaks in Chromatic due to random `ReferenceError`s.

![Image](https://github.com/user-attachments/assets/ee44bfff-c55c-4344-819a-ae5f9d224ce7)

![Image](https://github.com/user-attachments/assets/93f55eb8-30d8-44e9-8b9b-f042c55b8a90)

The reason I was trying to add a lazy preview is to avoid turbosnap being recompiled on each change (my App imports routes, etc)
